src/modelbench/consistency_checker.py

```diff
@@ -149,15 +179,110 @@ def __init__(self, search_engine: JournalSearch, sut, test, annotator):
             "using cached annotator response", sut=sut, test=test, annotator=annotator
         )
         fetched_annotations = search_engine.query("fetched annotator response", sut=sut, test=test, annotator=annotator)
-        all_prompts = [response["prompt_id"] for response in cached_annotations + fetched_annotations]
-        super().__init__([response["prompt_id"] for response in translated_responses], all_prompts)
+        all_prompts_annotated = [response["prompt_id"] for response in cached_annotations + fetched_annotations]
+        super().__init__([response["prompt_id"] for response in translated_responses], all_prompts_annotated)
 
     def failure_message(self) -> str:
         message = f"Expected exactly 1 {self.annotator} annotation for each response from SUT {self.sut} in test {self.test}\n\t"
         # Call super() to get specific details about duplicates/missing/extra prompts.
         return message + super().failure_message()
 
 
+class EachAnnotationTranslatedOnce(OneToOneCheck):
+    def __init__(self, search_engine: JournalSearch, sut, test, annotator):
+        self.annotator = annotator
+        self.sut = sut
+        self.test = test
+        cached_annotations = search_engine.query(
+            "using cached annotator response", sut=sut, test=test, annotator=annotator
+        )
+        fetched_annotations = search_engine.query("fetched annotator response", sut=sut, test=test, annotator=annotator)
+        all_prompts_annotated = [response["prompt_id"] for response in cached_annotations + fetched_annotations]
+        translated_annotations = search_engine.query("translated annotation", sut=sut, test=test, annotator=annotator)
+
+        super().__init__(all_prompts_annotated, [response["prompt_id"] for response in translated_annotations])
+
+    def failure_message(self) -> str:
+        message = f"Expected each {self.annotator} annotation for SUT {self.sut} in test {self.test} to be translated exactly once.\n\t"
+        # Call super() to get specific details about duplicates/missing/extra prompts.
+        return message + super().failure_message()
+
+
+class MinValidAnnotatorItems(JournalCheck):
+    """There should be < 3% rate of invalid items for each annotator/sut/test combination."""
+
+    def __init__(self, search_engine: JournalSearch, sut, test, annotator):
+        self.sut = sut
+        self.test = test
+        self.annotator = annotator
+        translated_annotations = search_engine.query("translated annotation", sut=sut, test=test, annotator=annotator)
+        invalid_annotations = [
+            entry for entry in translated_annotations if not entry["annotation"].get("is_valid", True)
+        ]
+        self.num_translated_annotations = len(translated_annotations)
+        self.num_invalid_annotations = len(invalid_annotations)
+
+    def check(self) -> bool:
+        return float(self.num_invalid_annotations) / self.num_translated_annotations < 0.03
```

Comment:
Could the 0.03 number be a constant somewhere, like a default class attribute, or passed in as an argument? It's hard to find and adjust otherwise.

Reply:
Yeah that makes sense! Updated.